configparser.go

```diff
@@ -140,6 +140,9 @@ func (c *ConfigParser) Read(r io.Reader) (err error) {
 				// in case of continuation line, value could be empty
 				if value != "" {
 					c.sections[curSect].options[key] = append(c.sections[curSect].options[key], value)
+				} else {
+					// ensure sure we do have have a key
+					c.sections[curSect].options[key] = append(c.sections[curSect].options[key])
```

Comment:
Hm, this append() has only a single argument, isn't that equivalent to `c.sections[curSect].options[key] = c.sections[curSect].options[key]` ? and (in most? all?) cases to `c.sections[curSect].options[key]  = nil` ?

Reply:
Sorry, it should be simpler being `c.sections[curSect].options[key] = c.sections[curSect].options[key]` yes.

It is never nil. It is empty list if not set.

Reply:
Also, why I do not know what this is supposed to mean the current code allows things like:

```
[section]
key=a
key=b
```

In that case, we get  `a\nb`.

If we used `c.sections[curSect].options[key] = []string{}` in the case `value == ""`, and if the input was:

```
[section]
key=a
key=
```

We would get an empty string. Whereas with `c.sections[curSect].options[key] = c.sections[curSect].options[key]`, we would get `a`, which would look more correct.

Though this is an undocumented and untested case. But the code still accepts it.